### PR TITLE
Task: Add redirect for popup

### PR DESCRIPTION
**Pull request by andrico1234**
Resolves: https://github.com/openui/open-ui/issues/634

Note: For some reason Yarn upgraded itself when I ran it on my machine. Would someone be able to pull my branch and run Yarn as well? I'm curious to see if it's going to cause issues on other devices

### CODE REVIEW

**Comment by andrico1234** [comment]
@mfreed7 @gregwhitworth 

This one's ready for review + testing

> Note: For some reason Yarn upgraded itself when I ran it on my machine. Would someone be able to pull my branch and run Yarn as well? I'm curious to see if it's going to cause issues on other devices

File: research/src/pages/components/popup.research.explainer.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { navigate } from 'gatsby'
+import { useEffect } from 'react'
+
+/**
+ *
+ * This component acts as a redirect to ensure backlinks aren't broken.
```
Comment (mfreed7) [inline research/src/pages/components/popup.research.explainer.js:6]: I suppose you could add a comment like

```
* This component acts as a redirect (from [1] to [2]) to ensure backlinks aren't broken.
* [1] https://open-ui.org/components/popup.research.explainer
* [2] https://open-ui.org/components/popover.research.explainer
```

File: research/src/pages/popup/popover.research.explainer.mdx [hunk 2]
```diff
@@ -49,26 +50,26 @@ Please also see the [WHATWG html spec PR for this proposal](https://github.com/w
...
- 1. Popovers always appear **on top of other page content**.
- 2. Popovers are **ephemeral**. When the user "moves on" to another part of the page (e.g. by clicking elsewhere, or hitting ESC), the popover hides.
- 3. Popovers (of a particular type) are generally **"one at a time"** - opening one popover closes others.
 
-This document proposes a set of APIs to make this type of UI easy to build.
+1.  Popovers always appear **on top of other page content**.
+2.  Popovers are **ephemeral**. When the user "moves on" to another part of the page (e.g. by clicking elsewhere, or hitting ESC), the popover hides.
+3.  Popovers (of a particular type) are generally **"one at a time"** - opening one popover closes others.
 
+This document proposes a set of APIs to make this type of UI easy to build.
```
Comment (mfreed7) [inline research/src/pages/popup/popover.research.explainer.mdx:58]: This looks better.

File: research/src/pages/popup/popover.research.explainer.mdx [hunk 3]
```diff
@@ -170,19 +172,19 @@ The declarative trigger attributes can also be accessed via IDL:
 
 ```javascript
 // These set the IDREF for the target element, and not the element itself:
-myButton.popoverToggleTarget = idref;
```
Comment (mfreed7) [inline research/src/pages/popup/popover.research.explainer.mdx:173]: Any reason to remove the `;`? I've always thought JS's optional `;` was a huge mistake.

File: research/src/pages/popup/popover.research.explainer.mdx [hunk 4]
```diff
@@ -194,43 +196,41 @@ There are several conditions that will cause `showPopover()` and/or `hidePopover
...
 4. Calling `hidePopover()` on a valid popover that is not currently showing. This will throw an `InvalidStateError` `DOMException`.
 
-
 ### Page Load Trigger
 
 As mentioned above, a `<div popover>` will be hidden by default. If it is desired that the popover should be shown automatically upon page load, the `defaultopen` attribute can be applied:
 
 ```html
-<div popover defaultopen>
+<div popover defaultopen></div>
```
Comment (mfreed7) [inline research/src/pages/popup/popover.research.explainer.mdx:204]: Really, the whole `defaultopen` section should be removed. But ok for now.

File: research/src/pages/popup/popover.research.explainer.mdx [hunk 5]
```diff
@@ -358,50 +356,47 @@ This not only allows developer ease-of-use from Javascript, but also allows for
...
 This allows feature detection of the values, for forward compatibility:
 
 ```javascript
 function supportsPopoverType(type) {
-  if (!HTMLElement.prototype.hasOwnProperty("popover"))
-    return false; // Popover API not supported
-  const testElement = document.createElement('div');
-  testElement.popover = type;
-  return testElement.popover == type.toLowerCase();
+  if (!HTMLElement.prototype.hasOwnProperty('popover')) return false // Popover API not supported
```
Comment (mfreed7) [inline research/src/pages/popup/popover.research.explainer.mdx:379]: Disagree with this one too - having a control flow statement that far from the front of the line is a bug waiting to happen.

File: research/src/pages/popup/popover.research.explainer.mdx [hunk 6]
```diff
@@ -625,8 +605,11 @@ Note that using the API described in this explainer, it is possible for elements
 
 ```html
 <my-popover>
-  <template shadowroot=closed>
-    <div popover=auto>This is a popover: <slot></slot></div>
+  <template shadowroot="closed">
```
Comment (mfreed7) [inline research/src/pages/popup/popover.research.explainer.mdx:608]: Given recent declarative shadow dom conversations, sounds like we should make this `<template shadowrootmode="closed">`. But again, for another day.

File: research/src/pages/popup/popover.research.explainer.mdx [hunk 7]
```diff
@@ -689,10 +671,13 @@ Similarly here, the developer is building **a menu**, so they should use the `<m
 In both cases, not only does the use of the semantically-correct element make it easier to understand the content, it also enforces the appropriate content rules and behaviors that are specific to each element. Further, using the correct element allows the UA to correctly represent the content in the accessibility tree, making assistive technologies better able to assist users in navigating the content.
 
 In an alternative proposal where the popover behavior is enabled via a special element, e.g. `<popup>`, all of the above would need to be carefully managed by the developer via ARIA roles and attributes, as there is no single role that can consistently be used to identify a `<popup>` element:
+
 ```html
-<popup role=dialog>
-<popup role=list>
-...etc...
+<popup role="dialog">
+  <popup role="list">
```
Comment (mfreed7) [inline research/src/pages/popup/popover.research.explainer.mdx:677]: I think maybe this should not be nested:

```
<popup role=dialog></popup>
<popup role=list></popup>
```

**Review by mfreed7 — APPROVED** [review]
Thanks for doing this!

Looks good to me. Thanks for the general cleanup. I added a few comments - generally I'm great with the cleanup, but I did disagree with a few of them.

You might also mention the cleanup in the commit description.

**Comment by andrico1234** [comment]
@mfreed7 , thanks for reviewing the PR

As for as the changes to "popover.research.explainer.mdx" go, I have a feeling that Prettier automatically formatted the file when I made a change to it. I'll revert any formatting only changes to reduce the change surface of this change

**Pull request merged**